tests/plotting/test_plot_manager.py

```diff
@@ -406,3 +407,57 @@ async def test_plot_info_caching(test_environment):
     await run_refresh_test(plot_manager)
     assert len(plot_manager.plots) == len(plot_manager.plots)
     plot_manager.stop_refreshing()
+
+
+@pytest.mark.parametrize(
+    ["event_to_raise"],
+    [
+        pytest.param(PlotRefreshEvents.started, id="started"),
+        pytest.param(PlotRefreshEvents.batch_processed, id="batch_processed"),
+        pytest.param(PlotRefreshEvents.done, id="done"),
+    ],
+)
+@pytest.mark.asyncio
+async def test_callback_event_raises(test_environment, event_to_raise: PlotRefreshEvents):
+    last_even_fired: Optional[PlotRefreshEvents] = None
+
+    def raising_callback(event: PlotRefreshEvents, _: PlotRefreshResult):
+        nonlocal last_even_fired
+        last_even_fired = event
+        if event == event_to_raise:
+            raise Exception(f"run_raise_in_callback {event_to_raise}")
+
+    env: TestEnvironment = test_environment
+    # Load dir_1
+    add_plot_directory(env.root_path, str(env.dir_1.path))
```

Comment:
Whats the benefit? From the docs: `If str or bytes is passed in, it is returned unchanged.`

I might miss something but the fact that a passed in `bytes` object is returned unchanged sounds weird to me.

Reply:
If you had the bytes `b"abc.txt"` would you expect to open the file `abc.txt` or `b'abc.txt'`?  `str()` will give you the latter.  Also note that while Windows uses UTF-16 (IIRC) as the native form of names on the filesystem, Linux uses bytes.  Often probably UTF-8 encoded, but bytes none the less.  So, if someone specifies `b"\x00FGH\xff"` as a 5 byte long path, it definitely shouldn't be converted to the 13 character string `"b'\x00FGH\xff'"`.

`str()` is a way to get a human readable form of an object, sort of.  `os.fspath()` is a way to take a path-like object and get the proper form of it for usage when accessing a filesystem.

The benefit is admittedly a bit abstract in most cases, especially since we use `pathlib.Path` objects.  One aspect, looking to the future, is that it leaves it open for new path-like objects to have a `str()` response that isn't necessarily the proper thing to use when trying to access the filesystem.  More generally it is about being able to handle all forms of paths including those that aren't valid Unicode without having to get into conditionally using `os.fsdecode()` and such nor ending up with strings that have surrogate escapes in them and are unprintable.

```pycon
>>> print(str(b"\xff".decode(errors="surrogateescape")))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'utf-8' codec can't encode character '\udcff' in position 0: surrogates not allowed
```

Reply:
I see, thanks for explaining. I changed it, see af91225355ad072a6c408300d61c68daf9117ecd, but i still wonder in which cases this becomes a real issue in practice and at the end `add_plot_directory` expects a `str` and converts back and forth with `Path` and `str`. Just saying using `str` here was obviously not the root of the _issue_.

```python
def add_plot_directory(root_path: Path, str_path: str) -> Dict:
    log.debug(f"add_plot_directory {str_path}")
    config = load_config(root_path, "config.yaml")
    if str(Path(str_path).resolve()) not in get_plot_directories(root_path, config):
        config["harvester"]["plot_directories"].append(str(Path(str_path).resolve()))
    save_config(root_path, "config.yaml", config)
    return config
```